pyscf/tdscf/rks.py

```diff
@@ -56,7 +56,7 @@ def gen_vind(self, mf=None):
 
         mol = mf.mol
         mo_coeff = mf.mo_coeff
-        assert mo_coeff.dtype == numpy.double
+        # assert mo_coeff.dtype == numpy.double
```

Comment:
You are right. The code assumes that `A-B` is diagonal, which assumes `(ia|jb) = (ia|bj)`. I added the assert back in the molecular rks code and made pbc rks calls this molecular code only for non-hybrid & Gamma-point. Otherwise, the pbc tda code will be used.